Diagnostics/FreeBusyChecker/FreeBusyChecker.ps1

```diff
@@ -0,0 +1,298 @@
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnPremise
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises OAuth or DAuth Availability Configurations.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org All
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises and Exchange Online OAuth or DAuth Availability Configurations.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnPremise -Auth OAuth
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises Availability OAuth Configurations
+#>
+
+# Exchange On-Premises
+#>
+#region Properties and Parameters
+
+#Requires -Module ExchangeOnlineManagement
+#Requires -Module ActiveDirectory
+
+[CmdletBinding(DefaultParameterSetName = "FreeBusyInfo_OP", SupportsShouldProcess)]
+
+param(
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [ValidateSet('DAuth', 'OAuth', 'All', '')]
+    [string[]]$Auth,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [ValidateSet('ExchangeOnPremise', 'ExchangeOnline')]
+    [string[]]$Org,
+    [Parameter(Mandatory = $true, ParameterSetName = "Help")]
+    [switch]$Help,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremisesUser,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnlineUser,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremDomain,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremEWSUrl,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremLocalDomain
+)
+begin {
+    . $PSScriptRoot\Functions\OnPremDAuthFunctions.ps1
+    . $PSScriptRoot\Functions\OnPremOAuthFunctions.ps1
+    . $PSScriptRoot\Functions\ExoDAuthFunctions.ps1
+    . $PSScriptRoot\Functions\ExoOAuthFunctions.ps1
+    . $PSScriptRoot\Functions\htmlContent.ps1
+    . $PSScriptRoot\Functions\hostOutput.ps1
+    . $PSScriptRoot\Functions\CommonFunctions.ps1
+} end {
+    $Script:countOrgRelIssues = (0)
+    $Script:WebServicesVirtualDirectory = $null
+    $Script:Server = hostname
+    $Script:startingDate = (Get-Date -Format yyyyMMdd_HHmmss)
+    $Script:htmlFile = "$PSScriptRoot\FBCheckerOutput_$($Script:startingDate).html"
+
+    CheckIfExchangeServer($Script:Server)
+    loadingParameters
+    #Parameter input
+
+    if (-not $OnlineUser) {
+        $Script:UserOnline = Get-RemoteMailbox -ResultSize 1 -WarningAction SilentlyContinue
+        $Script:UserOnline = $Script:UserOnline.RemoteRoutingAddress.SmtpAddress
+    } else {
+        $Script:UserOnline = Get-RemoteMailbox $OnlineUser -ResultSize 1 -WarningAction SilentlyContinue -ErrorAction SilentlyContinue
+        $Script:UserOnline = $Script:UserOnline.RemoteRoutingAddress.SmtpAddress
+    }
+
+    $Script:ExchangeOnlineDomain = ($Script:UserOnline -split "@")[1]
+
+    if ($Script:ExchangeOnlineDomain -like "*.mail.onmicrosoft.com") {
```

Comment:
What about other cloud? Is the script expected to work on these (e.g., GCC, Gallatin…)?

Reply:
Not sure if this is 100% required before we release. It would be nice yes, but I would rather get something out there and us create an issue for this and address it soon.

Reply:
It's okay to address this with a later update. We should mention in the documentation that the script works with WW cloud only.

Reply:
@MarcoLFrancisco let's add this to the doc that this only works for WW cloud only currently.